ReactNativeClient/lib/models/Setting.js

```diff
@@ -230,30 +245,51 @@ class Setting extends BaseModel {
 					return options;
 				},
 			},
+			themeAutoDetect: {
+				value: false,
+				type: Setting.TYPE_BOOL,
+				section: 'appearance',
+				appTypes: ['desktop'],
+				public: true,
+				label: () => _('Auto Detect'),
+				description: () =>
+					_('Enable to detect your system theme'),
+			},
 			theme: {
 				value: Setting.THEME_LIGHT,
 				type: Setting.TYPE_INT,
-				public: true,
+				public: false,
 				appTypes: ['mobile', 'desktop'],
 				isEnum: true,
 				label: () => _('Theme'),
 				section: 'appearance',
-				options: () => {
-					const output = {};
-					output[Setting.THEME_LIGHT] = _('Light');
-					output[Setting.THEME_DARK] = _('Dark');
-					if (platform !== mobilePlatform) {
-						output[Setting.THEME_DRACULA] = _('Dracula');
-						output[Setting.THEME_SOLARIZED_LIGHT] = _('Solarised Light');
-						output[Setting.THEME_SOLARIZED_DARK] = _('Solarised Dark');
-						output[Setting.THEME_NORD] = _('Nord');
-						output[Setting.THEME_ARITIM_DARK] = _('Aritim Dark');
-					} else {
-						output[Setting.THEME_OLED_DARK] = _('OLED Dark');
-					}
-					return output;
-				},
+				options: themeOptions,
+			},
+			preferredLightTheme: {
+				value: Setting.THEME_LIGHT,
+				type: Setting.TYPE_INT,
+				public: true,
+				appTypes: ['mobile', 'desktop'],
+				isEnum: true,
+				label: () => _('Preferred Light Theme'),
+				description: () =>
+					_('Applicable for Light System Theme and Night Mode button'),
+				section: 'appearance',
+				options: themeOptions,
+			},
+			preferredDarkTheme: {
+				value: Setting.THEME_DARK,
+				type: Setting.TYPE_INT,
+				public: true,
+				appTypes: ['mobile', 'desktop'],
+				isEnum: true,
+				label: () => _('Preferred Dark Theme'),
+				description: () =>
+					_('Applicable for Dark System Theme and Night Mode button'),
```

Comment:
'Applicable for Dark System Theme and Night Mode'

Reply:
Noted.